signature-aggregator/api/api.go

```diff
@@ -34,75 +36,128 @@ type SignatureAggregationByIDRequest struct {
+	UnsignedMessage string `json:"unsigned-message"`
 	// Optional hex or cb58 encoded signing subnet ID. If omitted will default to the subnetID of the source BlockChain
 	SigningSubnetID string `json:"signing-subnet-id"`
 	// Optional. Integer from 0 to 100 representing the percentage of the quorum that is required to sign the message
 	// defaults to 67 if omitted.
-	QuorumNum *uint64 `json:"quorum-num"`
+	QuorumNum uint64 `json:"quorum-num"`
 }
 
 type SignatureAggregationResponse struct {
-	// base64 encoding of the signature
-	SignedMessageBytes []byte `json:"signed-message-bytes"`
+	// hex encoding of the signature
+	SignedMessage string `json:"signed-message"`
 }
 
 func HandleSignatureAggregationRawRequest(logger logging.Logger, signatureAggregator *aggregator.SignatureAggregator) {
 	http.Handle(RawMessageAPIPath, signatureAggregationAPIHandler(logger, signatureAggregator))
 }
 
+func writeJsonError(
+	logger logging.Logger,
+	w http.ResponseWriter,
+	errorMsg string,
+) {
+	resp, err := json.Marshal(struct{ error string }{error: errorMsg})
+	if err != nil {
+		logger.Error(
+			"Error marshalling JSON error response",
+			zap.Error(err),
+		)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	w.Write(resp)
+	if err != nil {
+		logger.Error("Error writing error response", zap.Error(err))
+	}
+}
+
 func signatureAggregationAPIHandler(logger logging.Logger, aggregator *aggregator.SignatureAggregator) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		var req SignatureAggregationRawRequest
 		err := json.NewDecoder(r.Body).Decode(&req)
 		if err != nil {
-			logger.Warn("Could not decode request body")
-			http.Error(w, err.Error(), http.StatusBadRequest)
+			msg := "Could not decode request body"
+			logger.Warn(msg, zap.Error(err))
+			writeJsonError(logger, w, msg)
 			return
 		}
-		unsignedMessage, err := types.UnpackWarpMessage(req.UnsignedMessageBytes)
+		var decodedMessage []byte
+		decodedMessage, err = hex.DecodeString(
+			strings.TrimPrefix(req.UnsignedMessage, "0x"),
+		)
+		if err != nil {
+			msg := "Could not decode message"
+			logger.Warn(
+				msg,
+				zap.String("msg", req.UnsignedMessage),
+				zap.Error(err),
+			)
+			writeJsonError(logger, w, msg)
+			return
+		}
+		unsignedMessage, err := types.UnpackWarpMessage(decodedMessage)
 		if err != nil {
-			logger.Warn("Error unpacking warp message", zap.Error(err))
-			http.Error(w, err.Error(), http.StatusBadRequest)
+			msg := "Error unpacking warp message"
+			logger.Warn(msg, zap.Error(err))
+			writeJsonError(logger, w, msg)
 			return
 		}
 		var quorumNum uint64
-		if req.QuorumNum == nil {
+		if req.QuorumNum == 0 {
 			quorumNum = defaultQuorumNum
-		} else if *req.QuorumNum >= 0 || *req.QuorumNum > 100 {
-			logger.Warn("Invalid quorum number", zap.Uint64("quorum-num", *req.QuorumNum))
-			http.Error(w, "invalid quorum number", http.StatusBadRequest)
+		} else if req.QuorumNum >= 0 || req.QuorumNum > 100 {
```

Comment:
```suggestion
		} else if req.QuorumNum <= 0 || req.QuorumNum > 100 {
```

Reply:
The first portion shouldn't be necessary since it's uint right now either

my suggestion would be
```suggestion
		} else if req.QuorumNum > 100 {
```

Reply:
addressed in https://github.com/ava-labs/awm-relayer/pull/393/commits/da09584537decb4de155eb37c00ad64af61931a4